lib/libfrr.c

```diff
@@ -1101,9 +1101,11 @@ static void frr_terminal_close(int isexit)
 			     "%s: failed to open /dev/null: %s", __func__,
 			     safe_strerror(errno));
 	} else {
-		dup2(nullfd, 0);
-		dup2(nullfd, 1);
-		dup2(nullfd, 2);
+		int fd;
+		/* only redirect stdin, stdout, stderr to null when a tty */
+		for (fd = 2; fd >= 0; fd--)
+			if (isatty(fd))
```

Comment:
`isatty` is not the right thing to do here, because `daemon -d --log stdout` should work too, including if stdout is in fact a TTY.

The check needs to be `is fd X open as log target` instead.